src/udev/udev-builtin-hwdb.c

```diff
@@ -63,20 +64,18 @@ int udev_builtin_hwdb_lookup(struct udev_device *dev,
 
 static const char *modalias_usb(struct udev_device *dev, char *s, size_t size) {
         const char *v, *p;
-        int vn, pn;
+        uint16_t vn, pn;
 
         v = udev_device_get_sysattr_value(dev, "idVendor");
         if (!v)
                 return NULL;
         p = udev_device_get_sysattr_value(dev, "idProduct");
         if (!p)
                 return NULL;
-        vn = strtol(v, NULL, 16);
-        if (vn <= 0)
-                return NULL;
-        pn = strtol(p, NULL, 16);
-        if (pn <= 0)
-                return NULL;
+        if (safe_atoux16(v, &vn))
```

Comment:
Sorry, but this is still not it ;(

We only use bools and pointers like that in boolean context. Please use `if (safe_atoux16(v, &vn) < 0) ...`.

Reply:
Ah, I see. Thank you very much for guiding me through my first systemd patch!